Chap_Introduction.tex

```diff
@@ -389,5 +389,9 @@ \section{Version 3.1: Jan. 2019}
     \item Add new attributes to specify the level of information being requested where ambiguity may exist (see \ref{api:struct:attributes:retrieval})
     \item Add new attributes to assemble information by its level for storage where ambiguity may exist (see \ref{api:struct:attributes:storage})
     \item Add flag and \refmacro{PMIX_INFO_IS_END} macro for marking and detecting the end of a \refstruct{pmix_info_t} array
+    \item Remove \code{PMIX_NUM_SLOTS} as it is duplicative of (a) \refattr{PMIX_UNIV_SIZE} when used at the \refterm{session} level and (b) \refattr{PMIX_MAX_PROCS} when used at the \refterm{job} and \refterm{application} levels
```

Comment:
Isn't that a non-backward compatible change (although it is ABI compatible)?

Reply:
Fair point - suggestions? As I see it, we could:

- leave the duplication in and just add the explanation to the definition
- handle the conversion to/from PMIX_NUM_SLOTS in the library
- ??

The first option is obviously the easiest, but is that what we want to do?

Reply:
I would personally suggest to leave the duplication in v3.x and remove PMIX_NUM_SLOTS in v4.

Reply:
I have restored it, clarifying that it is duplicative. I did not deprecate it as it probably just isn't worth it.